### PR TITLE
initial version of defined resource types quest

### PR DESCRIPTION
**Pull request by kjhenner**

### CODE REVIEW

File: Quest_Guide/quests/defined_resource_types.md [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+
+- Understand how to manage multiple groups of resources with defined resource types.
+- Use a defined resource type to easily create hope pages for users.
+
+## Getting Started
+
+In the quest on parameterized classes, you saw how you can use parameters to
+customize a class as it is declared. If you recall that classes, like resources,
+can only be realized a single time in a given catalog, you might be wondering
+what to do if you Puppet to repeat the same pattern multiple times, but with
```
Comment (josephoaks) [inline Quest_Guide/quests/defined_resource_types.md]: please insert "want" between you + Puppet

**Comment by kjhenner** [comment]
Waiting on review for technical accuracy and completeness.

File: Quest_Guide/quests/defined_resource_types.md [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+
+You'll see the following:
+
+```
+location ~ ^/~(.+?)(/.*)?$ {
+    alias /home/$1/public_html$2;
+    autoindex on;
+}
+```
+
```
Comment (joshsamuelson) [inline Quest_Guide/quests/defined_resource_types.md]: Maybe add more explanation of how the regex rule works and a note about not needing to really understand it.

**Comment by joshsamuelson** [comment]
+1

**Comment by kjhenner** [comment]
Thanks @joshsamuelson. There are still a couple of things I want to work on before merging.

**Comment by joshsamuelson** [comment]
That helps a lot to add the explanation of the regex.  I think it's ready to merge from a technical perspective.  Needs a proofreading review from someone else.

**Pull request merged**